### ISSUE TITLE
Error Asset Report and Depreciation Report

### ISSUE THREAD

**Issue by rikysyaputra06** [snipe/snipe-it#1428]
#############  If i'am open on Depreciation Report = ################

21. Illuminate\View\Engines\CompilerEngine handleViewException
…\vendor\laravel\framework\src\Illuminate\View\Engines\PhpEngine.php41
20. ErrorException
…\app\storage\views\d9e8a0e9de1bd8f801e0247f5a2a92d356
19. Illuminate\Exception\Handler handleError
…\app\storage\views\d9e8a0e9de1bd8f801e0247f5a2a92d356
18. include
…\vendor\laravel\framework\src\Illuminate\View\Engines\PhpEngine.php37
17. Illuminate\View\Engines\PhpEngine evaluatePath
…\vendor\laravel\framework\src\Illuminate\View\Engines\CompilerEngine.php56
16. Illuminate\View\Engines\CompilerEngine get
…\vendor\laravel\framework\src\Illuminate\View\View.php134
15. Illuminate\View\View getContents
…\vendor\laravel\framework\src\Illuminate\View\View.php102
14. Illuminate\View\View renderContents
…\vendor\laravel\framework\src\Illuminate\View\View.php76
13. Illuminate\View\View render
…\vendor\laravel\framework\src\Illuminate\Http\Response.php43
12. Illuminate\Http\Response setContent
…\vendor\symfony\http-foundation\Symfony\Component\HttpFoundation\Response.php202
11. Symfony\Component\HttpFoundation\Response __construct
…\vendor\laravel\framework\src\Illuminate\Routing\Router.php1501
10. Illuminate\Routing\Router prepareResponse
…\vendor\laravel\framework\src\Illuminate\Routing\Router.php1036
9. Illuminate\Routing\Router dispatchToRoute
…\vendor\laravel\framework\src\Illuminate\Routing\Router.php1001
8. Illuminate\Routing\Router dispatch
…\vendor\laravel\framework\src\Illuminate\Foundation\Application.php775
7. Illuminate\Foundation\Application dispatch
…\vendor\laravel\framework\src\Illuminate\Foundation\Application.php745
6. Illuminate\Foundation\Application handle
…\vendor\barryvdh\laravel-debugbar\src\Middleware\Stack.php34
5. Barryvdh\Debugbar\Middleware\Stack handle
…\vendor\laravel\framework\src\Illuminate\Session\Middleware.php72
4. Illuminate\Session\Middleware handle
…\vendor\laravel\framework\src\Illuminate\Cookie\Queue.php47
3. Illuminate\Cookie\Queue handle
…\vendor\laravel\framework\src\Illuminate\Cookie\Guard.php51
2. Illuminate\Cookie\Guard handle
…\vendor\stack\builder\src\Stack\StackedHttpKernel.php23
1. Stack\StackedHttpKernel handle
…\vendor\laravel\framework\src\Illuminate\Foundation\Application.php641
0. Illuminate\Foundation\Application run
…\public\index.php49

########### If i'm open on Asset Report = #############

21. Illuminate\View\Engines\CompilerEngine handleViewException
…\vendor\laravel\framework\src\Illuminate\View\Engines\PhpEngine.php41
20. ErrorException
…\app\storage\views\eb92172c94d861a214185c67d354c5f558
19. Illuminate\Exception\Handler handleError
…\app\storage\views\eb92172c94d861a214185c67d354c5f558
18. include
…\vendor\laravel\framework\src\Illuminate\View\Engines\PhpEngine.php37
17. Illuminate\View\Engines\PhpEngine evaluatePath
…\vendor\laravel\framework\src\Illuminate\View\Engines\CompilerEngine.php56
16. Illuminate\View\Engines\CompilerEngine get
…\vendor\laravel\framework\src\Illuminate\View\View.php134
15. Illuminate\View\View getContents
…\vendor\laravel\framework\src\Illuminate\View\View.php102
14. Illuminate\View\View renderContents
…\vendor\laravel\framework\src\Illuminate\View\View.php76
13. Illuminate\View\View render
…\vendor\laravel\framework\src\Illuminate\Http\Response.php43
12. Illuminate\Http\Response setContent
…\vendor\symfony\http-foundation\Symfony\Component\HttpFoundation\Response.php202
11. Symfony\Component\HttpFoundation\Response __construct
…\vendor\laravel\framework\src\Illuminate\Routing\Router.php1501
10. Illuminate\Routing\Router prepareResponse
…\vendor\laravel\framework\src\Illuminate\Routing\Router.php1036
9. Illuminate\Routing\Router dispatchToRoute
…\vendor\laravel\framework\src\Illuminate\Routing\Router.php1001
8. Illuminate\Routing\Router dispatch
…\vendor\laravel\framework\src\Illuminate\Foundation\Application.php775
7. Illuminate\Foundation\Application dispatch
…\vendor\laravel\framework\src\Illuminate\Foundation\Application.php745
6. Illuminate\Foundation\Application handle
…\vendor\barryvdh\laravel-debugbar\src\Middleware\Stack.php34
5. Barryvdh\Debugbar\Middleware\Stack handle
…\vendor\laravel\framework\src\Illuminate\Session\Middleware.php72
4. Illuminate\Session\Middleware handle
…\vendor\laravel\framework\src\Illuminate\Cookie\Queue.php47
3. Illuminate\Cookie\Queue handle
…\vendor\laravel\framework\src\Illuminate\Cookie\Guard.php51
2. Illuminate\Cookie\Guard handle
…\vendor\stack\builder\src\Stack\StackedHttpKernel.php23
1. Stack\StackedHttpKernel handle
…\vendor\laravel\framework\src\Illuminate\Foundation\Application.php641
0. Illuminate\Foundation\Application run
…\public\index.php49

Can you help me with in the problems...

**Comment by snipe** [comment]
My guess would be that you have some bad data in your database. Did you manually edit any of the data in the database?

**Comment by rikysyaputra06** [comment]
I am not entirely manually change data in the database , I change his name url other than that there was nothing I fox .

**Comment by snipe** [comment]
I'd need to see the offending line in `\app\storage\views\d9e8a0e9de1bd8f801e0247f5a2a92d356`, but this still looks like bad data to me, as I can't reproduce it locally or on the demo.

**Comment by jayromems** [comment]
I'm getting a similar problem when simply running an Asset Report.

ErrorException (E_ERROR)
￼HELP￼
Trying to get property of non-object (View: /var/www/snipeit/app/views/backend/reports/asset.blade.php)
Open: /var/www/snipeit/vendor/laravel/framework/src/Illuminate/View/Engines/PhpEngine.php
		// an exception is thrown. This prevents any partial views from leaking.
		try
		{
			include $__path;
		}
		catch (\Exception $e)
		{
			$this->handleViewException($e, $obLevel);
		}
 
Could you please help?
Every other report seems ok. Mind you, I'm just getting started so I haven't populated the db with a whole lot yet.

**Comment by jayromems** [comment]
Nevermind, solved it on my own, DOH!
basically it wasn't working (and what you said above about manually editing the db tipped me off) 
I had removed the first location row in the db via phpmyadmin after I had created a new location. This meant that the new location row still had an id of '2' ... I guess the code still looks for an id of '1' when it does the parsing, hence the 'trying to get property of non-object'.
cheers
